storm-server/src/main/java/org/apache/storm/utils/ServerUtils.java

```diff
@@ -1217,9 +1199,127 @@ public static boolean areAllProcessesDead(Map<String, Object> conf, String user,
+            }
+            // check if existing process is owned by the specified user, if not, the process is dead
+            String pidUser;
+            try {
+                pidUser = Files.getOwner(pidDir.toPath()).getName();
+            } catch (NoSuchFileException ex) {
+                continue; // process died before the user can be checked
+            }
+            if (mockFileOwnerToUid) {
+                // code activated in testing to simulate Files.getOwner returning UID (which sometimes happens in runtime)
+                if (StringUtils.isNumeric(pidUser)) {
+                    LOG.info("Skip mocking, since owner {} of pidDir {} is already numeric", pidUser, pidDir);
+                } else {
+                    Integer uid = cachedUserToUidMap.get(pidUser);
+                    if (uid == null) {
+                        uid = ServerUtils.getUserId(pidUser);
+                        if (uid < 0) {
+                            String err = String.format("Cannot get UID for %s, while mocking the owner of pidDir %s",
+                                    pidUser, pidDir.getAbsolutePath());
+                            throw new IOException(err);
+                        }
+                        cachedUserToUidMap.put(pidUser, uid);
+                        LOG.info("Found UID {} for {}, while mocking the owner of pidDir {}", uid, pidUser, pidDir);
+                    } else {
+                        LOG.info("Found cached UID {} for {}, while mocking the owner of pidDir {}", uid, pidUser, pidDir);
+                    }
+                    pidUser = String.valueOf(uid);
+                }
+            }
+            //sometimes uid is returned instead of username - if so, try to convert and compare with uid
+            if (StringUtils.isNumeric(pidUser)) {
+                // numeric pidUser - this is UID
+                LOG.debug("Process directory {} owner is uid={}", pidDir, pidUser);
+                int pidUid = Integer.parseInt(pidUser);
+                Integer uid = cachedUserToUidMap.get(user);
+                if (uid == null) {
+                    uid = ServerUtils.getUserId(user);
+                    if (uid < 0) {
+                        String err = String.format("Cannot get uid for %s to compare with owner id=%d of process directory %s",
+                                user, pidUid, pidDir.getAbsolutePath());
+                        throw new IOException(err);
+                    }
+                    cachedUserToUidMap.put(user, uid);
+                }
+                if (uid == pidUid) {
+                    LOG.debug("Process {} is alive and owned by user {}/{}", pid, user, uid);
+                    return true;
+                } else {
+                    LOG.info("Prior process is dead, since directory {} owner {} is not same as expected user {}/{}, "
+                            + "likely pid {} was reused for a new process for uid {}", pidDir, pidUser, user, uid, pid, uid);
```

Comment:
Should the last "uid" be "pidUid"?

It is not easy to distinguish "user" vs "pidUser", "uid" vs "pidUid". I suggest to change them to be 
`actualUser vs expectedUser, actualUid vs expectedUid`

Reply:
renamed variables